### PR TITLE
Dark mode

### PR DESCRIPTION
**Pull request by vedanshgarg**

### CODE REVIEW

File: content/src/scripts/libs/contentDark.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
+// themeList.js
+
+const isPresentInThemeList = () => {
+  const domain = location.origin;
+  let themeList = JSON.parse(localStorage.getItem("spantree-themelist")) || {};
+
+  return domain in themeList && themeList[domain];
+};
+
```
Comment (tavyandy97) [inline content/src/scripts/libs/contentDark.js:9]: Can we remove the altered top nav bar height alteration. Causes the unecessary scrollbar introduction on smaller screens. (For ref I am talking when theme is switched from default to dark)

File: content/src/scripts/libs/contentDark.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
+// themeList.js
+
```
Comment (tavyandy97) [inline content/src/scripts/libs/contentDark.js:2]: State the purpose of this file in comments and why it was required to be executed as a different script. **Briefly**

File: content/src/scripts/libs/contentDark.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+darkGitlabTemp.type = "text/css";
+darkGitlabTemp.href = `${browserKey()}-extension://${chrome.i18n.getMessage(
+  "@@extension_id"
+)}/libs/gitlab-dark.css`;
+document
+  .querySelector("html")
+  .insertBefore(darkGitlabTemp, document.querySelector("html").childNodes[0]);
+
+
+document.addEventListener('DOMContentLoaded', fireContentLoadedEvent, false);
```
Comment (tavyandy97) [inline content/src/scripts/libs/contentDark.js:62]: Can this listener be removed, post execution of `fireContentLoadedEvent`? I know it won't be triggered more than once.

File: manifest.json [hunk 4]
```diff
@@ -16,6 +16,12 @@
       "matches": ["<all_urls>"],
```
Comment (tavyandy97) [inline manifest.json:16]: alter version to 0.0.2.3

File: content/src/scripts/libs/contentDark.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  // Remove CSS from Head
+
+  document.body.onload = () => {
+    document.querySelector("#spantree-theme-temp").remove();
+  };
+
+
+}
+
+
```
Comment (tavyandy97) [inline content/src/scripts/libs/contentDark.js:90]: remove extra spaces at the end of the file

File: content/src/scripts/libs/contentDark.js (hunk 3 above)
Comment (tavyandy97) [inline content/src/scripts/libs/contentDark.js:62]: Write the executional code at the end. Keep all the functions before the executional part.

File: content/src/scripts/libs/contentDark.js (hunk 2 above)
Comment (tavyandy97) [inline content/src/scripts/libs/contentDark.js:2]: Remove unnecessary spaces between code and comments. This file won't be minified.

**Comment by vedanshgarg** [comment]
> Screenshot_20200530_234910
> 
> Is there any way we can improve the scrollbar beside the collapsible navbar?

Yup, pushed a workaround for better look and user experience.

File: content/src/scripts/libs/contentDark.js (hunk 2 above)
Comment (vedanshgarg) [reply content/src/scripts/libs/contentDark.js:6]: Done

File: content/src/scripts/libs/contentDark.js (hunk 1 above)
Comment (vedanshgarg) [reply content/src/scripts/libs/contentDark.js]: the whole gitlab dark theme is based around the 50px topbar height and altering will require many changes to the css. Regarding the scrollbar introduction on smaller screen, I've pushed a change on scroll bars that might help.

Comment (tavyandy97) [reply content/src/scripts/libs/contentDark.js]: Works

**Pull request merged**